docker/user_service-ubuntu.sh

```diff
@@ -2,27 +2,46 @@
 [ -f .bashrc ] && . .bashrc
 set -ev
 INSTALL_DIR="${1:-/opt}"
+if [ -n "${PYTHON_EXECUTABLE}" ]; then 
+  PYTHON=$PYTHON_EXECUTABLE
+else
+  PYTHON='miniconda'
+fi
 
 # SIRF-Exercises
 git clone https://github.com/SyneRBI/SIRF-Exercises --recursive -b master $INSTALL_DIR/SIRF-Exercises
 
-if [ -f requirements-service.txt ]; then
-  conda install -c conda-forge -y --file requirements-service.txt || \
-  pip install -U -r requirements-service.txt
+if [ "$PYTHON" = "miniconda" ]; then
+  if [ -f requirements-service.yml ]; then
+    # installs the required packages in the environment with requirements-service.yml. 
+    # Notice that these requirements TEMPORARILY contains also the packages for SIRF-Exercises
+    mamba env update --file requirements-service.yml 
+  fi
+  conda clean -y --all
+
+# Python (runtime)
+else
+  if [ -f requirements-service.txt ]; then
+    pip install -U -r requirements-service.txt
+  fi
 fi
 
 #install SIRF-Exercises requirements
 cd $INSTALL_DIR/SIRF-Exercises
 if [ -f requirements.txt ]; then
-  conda install -c conda-forge -y --file requirements.txt || \
-  pip install -U -r requirements.txt
+  cat requirements.txt
+  python ~/install-sirf-exercises-dep.py requirements.txt
```

Comment:
This only makes sense when using conda/mamba. Otherwise we just have to use `pip`.
```suggestion
  if [ "$PYTHON" = "miniconda" ]; then
    # WARNING: temporary work-around for requirements not available in conda
    # This will only work when building the docker image as we rely on an absolute path here.
    python ~/install-sirf-exercises-dep.py requirements.txt
  else
    $PYTHON_EXECUTABLE -m pip install -U -r requirements.txt
  fi
```